internal/databases/postgres/connect.go

```diff
@@ -26,7 +26,15 @@ func Connect(configOptions ...func(config *pgx.ConnConfig) error) (*pgx.Conn, er
 	}
 	conn, err := pgx.Connect(config)
 	if err != nil {
-		if config.Host != "localhost" {
+		config.RuntimeParams["gp_role"] = "utility"
```

Comment:
I think it is OK to leave this hinge at the MVP stage, but I suggest extracting this logic to the explicit function to make it clear what is going on. For example, you may name it `tryConnectToGpSegment` or something like that. And leave some comment regarding this logic.